### PR TITLE
simpler code

### PR DESCRIPTION
**Pull request by joamatab**

### CODE REVIEW

File: femwell/tests/test_mesh.py [hunk 1]
```diff
@@ -47,7 +47,6 @@ def test_shared_edge():
                             offset_core = 0
                             )
     mesh = mesh_from_OrderedDict(shapes, resolutions = {})
-    assert True
```
Comment (HelgeGehring) [inline femwell/tests/test_mesh.py:50]: @simbilod these asserts are important to make the test run, right?

File: femwell/waveguide.py [hunk 2]
```diff
@@ -46,8 +46,16 @@ def mesh_waveguide(filename, wsim, hclad, hbox, wcore, hcore):
...
-        mesh = Mesh.load(tmpdirname + '/mesh.msh')
+        mesh = mesh_waveguide(
+            wsim=2,
+            hclad=0.7,
+            hbox=0.5,
+            wcore=0.5,
+            hcore=0.22,
+            filename=f'{tmpdirname}/mesh.msh',
+        )
+
```
Comment (HelgeGehring) [inline femwell/waveguide.py:57]:
```suggestion
        mesh = mesh_waveguide(wsim=2, hclad=.7, hbox=.5, wcore=0.5, hcore=0.22, filename=f'{tmpdirname}/mesh.msh')
```
I'd guess breaking the parameters into lines would lower the readability, as the file focusses on implementing the simulation and the parameters for the example are not in focus here

File: docs/source/photonics/examples/vary_width.py [hunk 3]
```diff
@@ -30,8 +30,14 @@
...
         with tempfile.TemporaryDirectory() as tmpdirname:
-            mesh_from_OrderedDict(polygons, resolutions, filename=tmpdirname + '/mesh.msh', default_resolution_max=.6)
-            mesh = Mesh.load(tmpdirname + '/mesh.msh')
+            mesh_from_OrderedDict(
+                polygons,
+                resolutions,
+                filename=f'{tmpdirname}/mesh.msh',
+                default_resolution_max=0.6,
+            )
+
```
Comment (HelgeGehring) [inline docs/source/photonics/examples/vary_width.py:39]:
```suggestion
            mesh_from_OrderedDict(polygons, resolutions, filename=f'{tmpdirname}/mesh.msh', default_resolution_max=.6)
```
The usage of f-strings is nice! I don't see the point in adding the line breaks here.

File: docs/source/photonics/examples/waveguide_modes.py [hunk 4]
```diff
@@ -25,8 +25,14 @@
...
 with tempfile.TemporaryDirectory() as tmpdirname:
-    mesh_from_OrderedDict(polygons, resolutions, filename=tmpdirname + '/mesh.msh', default_resolution_max=10)
-    mesh = Mesh.load(tmpdirname + '/mesh.msh')
+    mesh_from_OrderedDict(
+        polygons,
+        resolutions,
+        filename=f'{tmpdirname}/mesh.msh',
+        default_resolution_max=10,
+    )
+
```
Comment (HelgeGehring) [inline docs/source/photonics/examples/waveguide_modes.py:34]:
```suggestion
    mesh_from_OrderedDict(polygons, resolutions, filename=f'{tmpdirname}/mesh.msh', default_resolution_max=10)

```
The usage of f-strings is nice! I don't see the point in adding the line breaks here.

File: femwell/culomb.py [hunk 5]
```diff
@@ -57,9 +57,15 @@ def coulomb(u, v, w):
...
 
     with tempfile.TemporaryDirectory() as tmpdirname:
-        mesh_from_OrderedDict(polygons, resolutions, filename=tmpdirname + '/mesh.msh', default_resolution_max=5)
+        mesh_from_OrderedDict(
+            polygons,
+            resolutions,
+            filename=f'{tmpdirname}/mesh.msh',
+            default_resolution_max=5,
+        )
+
```
Comment (HelgeGehring) [inline femwell/culomb.py:66]:
```suggestion
        mesh_from_OrderedDict(polygons, resolutions, filename=f'{tmpdirname}/mesh.msh', default_resolution_max=5)
```
The usage of f-strings is nice! I don't see the point in adding the line breaks here.

File: femwell/eme.py [hunk 6]
```diff
@@ -281,8 +281,7 @@ def compute_total_S_matrix(meshnames, mesh_info_dict, lengths, wavelength, num_m
         background_padding=(2.0, 2.0, 2.0, 2.0), # how much of backgorund tag to add to each side of the simulatoin
         filename=f"mesh_x_{x1}.msh",
     ))
-    meshnames.append(f"mesh_x_{x1}.msh")
-
+    meshnames = [f"mesh_x_{x1}.msh"]
```
Comment (HelgeGehring) [inline femwell/eme.py:284]: I'd guess this is already prepared for being put in a loop? @simbilod Therefore, I'd suggest not changing this?
```suggestion
    meshnames.append(f"mesh_x_{x1}.msh")
```

File: femwell/eme.py [hunk 7]
```diff
@@ -256,18 +257,17 @@ def compute_total_S_matrix(meshnames, mesh_info_dict, lengths, wavelength, num_m
...
+        "core": {"resolution": 0.02, "distance": 2},
+        "slab90": {"resolution": 0.05, "distance": 2},
+        "Oxide": {"resolution": 0.3, "distance": 2},
+    }
 
     ymin = -6 # make sure all objects cross the line
     ymax = 6 # make sure all objects cross the line
 
     meshes = []
     lengths = []
```
Comment (HelgeGehring) [inline femwell/eme.py:270]:
```suggestion
    lengths = []
    meshnames = []
```
I'd guess this is already prepared for being put in a loop? @simbilod Therefore, I'd suggest not changing this?

File: femwell/examples/overlap_integrals.py [hunk 8]
```diff
@@ -15,8 +15,16 @@
...
-        mesh = Mesh.load(tmpdirname + '/mesh.msh')
+        mesh = mesh_waveguide(
+            wsim=2,
+            hclad=0.7,
+            hbox=0.5,
+            wcore=width,
+            hcore=0.22,
+            filename=f'{tmpdirname}/mesh.msh',
+        )
+
```
Comment (HelgeGehring) [inline femwell/examples/overlap_integrals.py:26]:
```suggestion
mesh = mesh_waveguide(wsim=2, hclad=.7, hbox=.5, wcore=width, hcore=0.22, filename=f'{tmpdirname}/mesh.msh')
```
I'd guess breaking the parameters into lines would lower the readability, as the file focuses on implementing the simulation and the parameters for the example are not in focus here. The usage of f-strings is nice!

**Review by HelgeGehring — CHANGES_REQUESTED** [review]
Many of the suggested code changes improve the code quality, thanks!

Some of them can be improved a bit more and some changes rather lower the code quality, I've made some suggestions :)

@simbilod could you have a look on the changes regarding the meshing?

**Comment by simbilod** [comment]
On the gdsfactory version of this I just accept all the reformatting, since it usually doesn't break the logic, it's less effort, and readability remains OK

In the end I'm also still not fully 100% happy with this meshing code, because complete 3D functionality will require more constructive geometry at the gmsh level. This will simplify the 2D version too.

**Comment by HelgeGehring** [comment]
@simbilod 
Okay, then I'll do the same :)
I'm only worried about 
```
        for _ in shapely_xy_polygon.geoms if hasattr(shapely_xy_polygon, 'geoms') else [shapely_xy_polygon]:
```
Is it intentional, that the variable it loops over is not used at all?

File: femwell/mesh/meshtracker.py [hunk 9]
```diff
@@ -155,7 +146,7 @@ def add_xy_surface(self, shapely_xy_polygon, label=None, physical=True):
         hole_loops = []
         surfaces_to_label = []
 
-        for polygon in shapely_xy_polygon.geoms if hasattr(shapely_xy_polygon, 'geoms') else [shapely_xy_polygon]:
+        for _ in shapely_xy_polygon.geoms if hasattr(shapely_xy_polygon, 'geoms') else [shapely_xy_polygon]:
```
[the remark this answers is not in the document]
Comment (HelgeGehring) [reply femwell/mesh/meshtracker.py:146]: might it be that the code within the loop should use `polygon` instead of `shapely_xy_polygon`?

**Pull request merged**